src/pudl/extract/ferc1.py

```diff
@@ -358,9 +358,9 @@ def create_raw_ferc1_assets() -> list[SourceAsset]:
     )
     dbf_table_names = tuple(set(flattened_dbfs))
     raw_ferc1_dbf_assets = [
-        SourceAsset(
+        AssetSpec(
```

Comment:
According to @bendnorman's investigation in #3977 it looks like you need to do something like

```python
AssetSpec(key=AssetKey("...")).with_io_manager_key("...")
```

Instead of adding the `io_manager_key` to the `metadata` dictionary.

Does that address the issues you are having with `AssetSpec` configuration you were mentioning?